CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword

```

Comment:
+1 and I would try to do something like this with every rule when possible:

-----
Rule : "Use built-in type alias"
// Good example
// Bad Example
Related .editorconfig settings: `dotnet_style_predefined_type_for_locals_parameters_members = true:warning`
StyleCop Rule: [SA1121](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1121.md) 

-----
I know it's a lot of work, but then we can easily customize automatic static code analysis, by enabling/disabling rules.

(The end result of this guide should be pre-defined .editorconfig, .vssetting and .ruleset files.